grails-app/domain/org/pih/warehouse/inventory/CycleCount.groovy

```diff
@@ -23,10 +25,35 @@ class CycleCount {
     def beforeInsert() {
         createdBy = AuthService.currentUser
         updatedBy = AuthService.currentUser
+        status = recomputeStatus()
     }
 
     def beforeUpdate() {
         updatedBy = AuthService.currentUser
+        status = recomputeStatus()
```

Comment:
Is it updated after related items change as well? Have you considered using a formula for it (see point 8.2.11 here https://grails.github.io/legacy-gorm-doc/6.0.x/hibernate/manual/index.html#ormdsl)? Perhaps it would be a good/suiting approach as well?

Reply:
The problem with having it be derived was that it was inaccessible when building the session view in cycle-count-session.sql. I didn't try it the way you suggest but I feel like that would have the same issue. I can test though

Reply:
Ok, I think this makes sense then.